### PR TITLE
added board configuration for Sparkfun ThingPlus RP2040

### PR DESCRIPTION
**Pull request by mintakka**
Added board configuration files for the [Sparkfun ThingPlus RP2040](https://learn.sparkfun.com/tutorials/rp2040-thing-plus-hookup-guide/hardware-overview)

Builtin NeoPixel, LED, QWIIC port and build-in micro-SD slot all tested and working with the updated pin definitions.

### CODE REVIEW

**Comment by falkenad** [comment]
@mintakka Shouldn't the flash size be 16 as they're 25Q128 flash chips?
IE: #define TOTAL_FLASH_SIZE 8 * 1024 * 1024
Should be:
#define TOTAL_FLASH_SIZE 16 * 1024 * 1024  ?

**Comment by mintakka** [comment]
Yes it should be, I missed that!

File: ports/raspberrypi/boards/sparkfun_thingplus/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#define DEFAULT_I2C_BUS_SDA (&pin_GPIO6)
+
+#define DEFAULT_SPI_BUS_SCK (&pin_GPIO14)
+#define DEFAULT_SPI_BUS_MOSI (&pin_GPIO15)
+#define DEFAULT_SPI_BUS_MISO (&pin_GPIO12)
+
+#define DEFAULT_UART_BUS_RX (&pin_GPIO1)
+#define DEFAULT_UART_BUS_TX (&pin_GPIO0)
+
+// Flash chip is GD25Q64 connected over QSPI
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_thingplus/mpconfigboard.h:16]:
```suggestion
// Flash chip is GD25Q128 connected over QSPI
```

File: ports/raspberrypi/boards/sparkfun_thingplus/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+USB_VID = 0x239A
+USB_PID = 0x80F2
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_thingplus/mpconfigboard.mk:2]:
```suggestion
USB_VID = 0x1B4F
USB_PID = 0x0025
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the definition! Please rename the folder to `sparkfun_thingplus_rp2040` because thingplus is a name used for the form factor. You'll also need to add the board to .github/workflows/build.yml

**Comment by mintakka** [comment]
Thank you, I will make these changes and re-submit

**Review by mintakka — COMMENTED** [review]
revised pull request with requested changes

**Comment by mintakka** [comment]
@tannewt - I've addressed the required changes and I think it is ready for re-review

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -322,6 +322,7 @@ jobs:
         - "sparkfun_samd21_dev"
         - "sparkfun_samd21_mini"
         - "sparkfun_samd51_thing_plus"
+        - "sparkfun_thingplus_rp2040"
```
Comment (tannewt) [inline .github/workflows/build.yml:325]:
```suggestion
        - "sparkfun_thing_plus_rp2040"
```

File: ports/raspberrypi/boards/sparkfun_thingplus_rp2040/mpconfigboard.mk [hunk 4]
```diff
@@ -0,0 +1,9 @@
+USB_VID = 0x1B4F
+USB_PID = 0x0025
+USB_PRODUCT = "ThingPlus RP2040"
+USB_MANUFACTURER = "Sparkfun"
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_thingplus_rp2040/mpconfigboard.mk:4]:
```suggestion
USB_MANUFACTURER = "SparkFun"
```

File: ports/raspberrypi/boards/sparkfun_thingplus_rp2040/mpconfigboard.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#define MICROPY_HW_BOARD_NAME "Sparkfun ThingPlus RP2040"
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_thingplus_rp2040/mpconfigboard.h:1]:
```suggestion
#define MICROPY_HW_BOARD_NAME "SparkFun Thing Plus - RP2040"
```

**Review by tannewt — CHANGES_REQUESTED** [review]
A few minor things. Sorry I didn't see them in the first round of reviewing.

**Review by mintakka — COMMENTED** [review]
All change requests incorporated

**Comment by mintakka** [comment]
Those are great catches, thanks!

**Review by mintakka — COMMENTED** [review]
aligned board name in boards.yml with the folder name for the board in ports/raspberrypi/boards

**Comment by mintakka** [comment]
Alright, looks like the build checks all passed now. I think this is ready for review.

**Review by jepler — APPROVED** [review]
Thank you!  Github tells me this is your first pull request to any repo, congratulations on that as well.

**Comment by mintakka** [comment]
awesome, thanks!

**Pull request merged**